From: win7-support-bot <win7@cromite.org>
Date: Sun, 26 Jul 2026 00:00:00 +0000
Subject: WIN7: chrome-browser / com

Files:
 chrome/browser/chrome_browser_main_win.cc

License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
diff --git a/chrome/browser/chrome_browser_main_win.cc b/chrome/browser/chrome_browser_main_win.cc
index 2d34ec6143a08..faebf636f3377 100644
--- a/chrome/browser/chrome_browser_main_win.cc
+++ b/chrome/browser/chrome_browser_main_win.cc
@@ -41,6 +41,7 @@
 #include "base/win/elevation_util.h"
 #include "base/win/pe_image.h"
 #include "base/win/win_util.h"
+#include "base/win/windows_version.h"
 #include "base/win/wrapped_window_proc.h"
 #include "build/branding_buildflags.h"
 #include "build/build_config.h"
@@ -552,12 +553,16 @@ void ChromeBrowserMainPartsWin::PostBrowserStart() {
 #endif
 
   // Record Processor Metrics. This is very low priority, hence posting as
-  // BEST_EFFORT to start after Chrome startup has completed.
-  scoped_refptr<base::SequencedTaskRunner> task_runner =
-      base::ThreadPool::CreateSequencedTaskRunner(
-          {base::MayBlock(), base::TaskPriority::BEST_EFFORT});
-  task_runner->PostTask(FROM_HERE,
-                        base::BindOnce(&DelayedRecordProcessorMetrics));
+  // BEST_EFFORT to start after Chrome startup has completed. This metric is
+  // only available starting Windows 10.
+  if (base::win::OSInfo::GetInstance()->version() >=
+      base::win::Version::WIN10) {
+    scoped_refptr<base::SequencedTaskRunner> task_runner =
+        base::ThreadPool::CreateSequencedTaskRunner(
+            {base::MayBlock(), base::TaskPriority::BEST_EFFORT});
+    task_runner->PostTask(FROM_HERE,
+                          base::BindOnce(&DelayedRecordProcessorMetrics));
+  }
 
   // Write current executable path to the User Data directory to inform
   // Progressive Web App launchers, which run from within the User Data
